src/stepfunctions/steps/utils.py

```diff
@@ -45,3 +46,24 @@ def get_aws_partition():
             return cur_partition
 
     return cur_partition
+
+
+def merge_dicts(first, second, first_name, second_name):
+    """
+    Merges first and second dictionaries into the first one.
+    Values in the first dict are updated with the values of the second one.
+    """
+    if all(isinstance(d, dict) for d in [first, second]):
+        for key, value in second.items():
+            if key in first:
+                if isinstance(first[key], dict) and isinstance(second[key], dict):
+                    merge_dicts(first[key], second[key], first_name, second_name)
+                elif first[key] == value:
+                    pass
+                else:
+                    logger.info(
+                        f"{first_name} property: <{key}> with value: <{first[key]}>"
+                        f" will be overwritten with value provided in {second_name} : <{value}>")
```

Comment:
question: Do we think this is useful? If not, can just use Python's built-in [`dict.update`](https://docs.python.org/3/library/stdtypes.html#dict.update)

Reply:
The built in update() does not take into account nested dictionary values - for ex:
```
d1 = {'a': {'aa': 1, 'bb': 2, 'c': 3}}
d2 = {'a': {'bb': 1}}

d1.update(d2)
print(d1)
```
Will have followint output: `{'a': {'bb': 1}}`

Reply:
Initially added them to facilitate troubleshooting, but I'm open to remove the logs if we deem them not useful enough or too noisy

Reply:
If the expected behaviour is well documented it seems unnecessary. Since the method only exists for logging, if we get rid of it there's less code to maintain. What do you think, @shivlaks?

Reply:
> The built in update() does not take into account nested dictionary values

Missed this comment. Since we need a deep merge, dict.update is not going to work here